src/common/src/array/list_array.rs

```diff
@@ -503,6 +504,42 @@ impl Debug for ListRef<'_> {
     }
 }
 
+impl ToText for ListRef<'_> {
+    fn to_text(&self) -> String {
+        iter_elems_ref!(self, it, {
+            format!(
+                "{{{}}}",
+                it.format_with(",", |datum_ref, f| {
+                    let s = datum_ref.to_text();
```

Comment:
Feels like duplicated with `Display` below except the usage of `to_text` here. However, there is no need to refactor and share code between these two. I actually found it is not hard to remove `Display` (as suggested in my previous comment). So this intermediate state is acceptable from my point of view.